internal/controller/postgrescluster/instance.go

```diff
@@ -342,6 +381,65 @@ func (r *Reconciler) observeInstances(
 	return observed, err
 }
 
+// storeDesiredRequest saves the appropriate request value to the PostgresCluster
+// status. If the value has grown, create an Event.
+func (r *Reconciler) storeDesiredRequest(
+	ctx context.Context, cluster *v1beta1.PostgresCluster,
+	instanceSetName, desiredRequest, desiredRequestBackup string,
+) string {
+	var current resource.Quantity
+	var previous resource.Quantity
+	var err error
+	log := logging.FromContext(ctx)
+
+	// Parse the desired request from the cluster's status.
+	if desiredRequest != "" {
+		current, err = resource.ParseQuantity(desiredRequest)
+		if err != nil {
+			log.Error(err, "Unable to parse pgData volume request from status ("+
+				desiredRequest+") for "+cluster.Name+"/"+instanceSetName)
+			// If there was an error parsing the value, treat as unset (equivalent to zero).
+			desiredRequest = ""
+			current, _ = resource.ParseQuantity("")
+
+		}
+	}
+
+	// Parse the desired request from the status backup.
+	if desiredRequestBackup != "" {
+		previous, err = resource.ParseQuantity(desiredRequestBackup)
+		if err != nil {
+			log.Error(err, "Unable to parse pgData volume request from status backup ("+
+				desiredRequestBackup+") for "+cluster.Name+"/"+instanceSetName)
+			// If there was an error parsing the value, treat as unset (equivalent to zero).
+			desiredRequestBackup = ""
+			previous, _ = resource.ParseQuantity("")
+
+		}
+	}
+
+	// Determine if the limit is set for this instance set.
+	var limitSet bool
+	for _, specInstance := range cluster.Spec.InstanceSets {
+		if specInstance.Name == instanceSetName {
+			limitSet = !specInstance.DataVolumeClaimSpec.Resources.Limits.Storage().IsZero()
+		}
+	}
+
+	if limitSet && current.Value() > previous.Value() {
+		r.Recorder.Eventf(cluster, corev1.EventTypeNormal, "VolumeAutoGrow",
+			"pgData volume expansion to %v requested for %s/%s.",
+			current.String(), cluster.Name, instanceSetName)
+	}
+
+	// If the desired size was not observed, update with previously stored value.
```

Comment:
:question: What causes desired size not to be observed, when `desiredRequestBackup` is set?

Reply:
In a scenario where the annotation on the Pod was missing. The easiest example is when the cluster is shutdown, but it could also happen if a Pod had been restarted and the annotation had not yet been updated.

Reply:
Can that be added to the comment?

Reply:
Sure :+1:

Reply:
Thanks!